### PR TITLE
TypeScript documentation

### PR DESCRIPTION
**Pull request by RainerKlute**

### CODE REVIEW

**Comment by terfloth** [comment]
@dTrace82 : please review

**Comment by andreasmuelder** [comment]
@dTrace82 what is the state here?

**Comment by RainerKlute** [comment]
I'd say it's not a final state. ;-)

File: plugins/org.yakindu.base.types/plugin.xml [hunk 1]
```diff
@@ -12,4 +12,12 @@
...
       </package>
    </extension>
+   <extension point="org.eclipse.emf.ecore.generated_package">
+      <!-- @generated base -->
+      <package
+            uri="http://www.yakindu.org/base/base/2.0.0"
+            class="org.yakindu.base.base.BasePackage"
+            genModel="model/base.genmodel"/>
+   </extension>
+
```
Comment (RainerKlute) [inline plugins/org.yakindu.base.types/plugin.xml]: This change has nothing to do with documentation and should thus not go here.

File: plugins/org.yakindu.base.types/src-gen/org/yakindu/base/base/impl/BaseFactoryImpl.java [hunk 2]
```diff
@@ -11,6 +11,7 @@
 import org.eclipse.emf.ecore.EPackage;
 import org.eclipse.emf.ecore.impl.EFactoryImpl;
 import org.eclipse.emf.ecore.plugin.EcorePlugin;
+import org.yakindu.base.base.*;
```
Comment (RainerKlute) [inline plugins/org.yakindu.base.types/src-gen/org/yakindu/base/base/impl/BaseFactoryImpl.java]: This change has nothing to do with documentation and should thus not go here.

File: plugins/org.yakindu.base.types/src-gen/org/yakindu/base/base/util/BaseAdapterFactory.java [hunk 3]
```diff
@@ -10,6 +10,7 @@
 import org.eclipse.emf.common.notify.Notifier;
 import org.eclipse.emf.common.notify.impl.AdapterFactoryImpl;
 import org.eclipse.emf.ecore.EObject;
+import org.yakindu.base.base.*;
```
Comment (RainerKlute) [inline plugins/org.yakindu.base.types/src-gen/org/yakindu/base/base/util/BaseAdapterFactory.java]: This change has nothing to do with documentation and should thus not go here.

File: plugins/org.yakindu.base.types/src-gen/org/yakindu/base/base/util/BaseSwitch.java [hunk 4]
```diff
@@ -9,6 +9,7 @@
 import org.eclipse.emf.ecore.EObject;
 import org.eclipse.emf.ecore.EPackage;
 import org.eclipse.emf.ecore.util.Switch;
+import org.yakindu.base.base.*;
```
Comment (RainerKlute) [inline plugins/org.yakindu.base.types/src-gen/org/yakindu/base/base/util/BaseSwitch.java]: This change has nothing to do with documentation and should thus not go here.

**Pull request merged**